src/BufferStream.ts

```diff
@@ -237,38 +237,39 @@ export class BufferStream {
         return hexString.toUpperCase();
     }
 
+    checkSize(step: number): void {
+        let size = this.buffer.byteLength;
+        if (this.offset + step < size) {
+            return
+        }
 
-    checkSize(step) {
-        while (this.offset + step > this.buffer.byteLength) {
-            //throw new Error("Writing exceeded the size of buffer");
-            //resize
-            var dst = new ArrayBuffer(this.buffer.byteLength * 2);
-            new Uint8Array(dst).set(new Uint8Array(this.buffer));
-            this.buffer = dst;
-            this.view = new DataView(this.buffer);
+        while (this.offset + step > size) {
+            size *= 2;
         }
+        var dst = new ArrayBuffer(size);
+        new Uint8Array(dst).set(new Uint8Array(this.buffer));
+        this.size = size
```

Comment:
I think this line is wrong but I can't be sure. `this.size` appears to refer to bytes actually written? It's increased in `increment` when data's actually been written to that length. This here would result in the buffer's apparent size being expanded to the new capacity.